include/RevProc.h

```diff
@@ -167,23 +169,26 @@ class RevProc{
   uint64_t ExecPC;          ///< RevProc: executing PC
   uint16_t HartToDecode;    ///< RevProc: Current executing ThreadID
   uint16_t HartToExec;      ///< RevProc: Thread to dispatch instruction
+  std::vector<std::shared_ptr<RevHart>> Harts; ///< RevProc: vector of Harts
   uint64_t Retired;         ///< RevProc: number of retired instructions
 
+  unsigned numHarts;        ///< RevProc: Number of Harts for this core
```

Comment:
You used `uint16_t` elsewhere; best to be consistent; if you think it might change, then
```c++
   using numHarts_t = uint16_t;
```
can be defined in a header file, and then everywhere a `numHarts` appears, it should be declared `numHarts_t`.

(Note: `using type1 = type2` is preferred to `typedef type2 type1` in C++11 and later. It can also be templatized.)

Reply:
+1, use the same type of the `Id` as well.  It's defined as an `unsigned` but printed as a `uint16_t`